packages/deployer/subtasks/validate-router.js

```diff
@@ -1,57 +1,44 @@
-const fs = require('fs');
-const path = require('path');
 const { subtask } = require('hardhat/config');
 const logger = require('@synthetixio/core-js/utils/logger');
-const relativePath = require('@synthetixio/core-js/utils/relative-path');
-const { getSelectors } = require('@synthetixio/core-js/utils/contracts');
-const filterValues = require('filter-values');
-const { SUBTASK_VALIDATE_ROUTER } = require('../task-names');
+const { getSourcesAST } = require('../internal/ast/ast-sources');
+const {
+  findDuplicateSelectorsInAST,
+  findMissingSelectorsInAST,
+  findUnreachableSelectorsInAST,
+} = require('../internal/ast/router-validator');
+const {
+  findMissingSelectorsInSource,
+  findRepeatedSelectorsInSource,
+  findWrongSelectorsInSource,
+} = require('../internal/router-source-validator');
+
+const { SUBTASK_VALIDATE_ROUTER, SUBTASK_CANCEL_DEPLOYMENT } = require('../task-names');
 
 subtask(
   SUBTASK_VALIDATE_ROUTER,
   'Runs a series of validations against a generated router source.'
 ).setAction(async (_, hre) => {
   logger.subtitle('Validating router');
 
-  const routerPath = path.join(
-    relativePath(hre.config.paths.sources, hre.config.paths.root),
-    'Router.sol'
-  );
+  const { contracts } = await getSourcesAST(hre);
 
-  const modules = filterValues(hre.deployer.deployment.general.contracts, (c) => c.isModule);
-  const modulesNames = Object.keys(modules);
+  let errorsFound = [];
```

Comment:
errorsFound will collect all the errors of the validations to log at the end of the process. Removed all logging from the validators.